AGGREGATE.md

```diff
@@ -1,180 +1,359 @@
+
+**Note:** as new and improved protocols are invented, we can consider adding them to the API in some form, this includes other aggregation functions like using larger domains (> 2^32) with some thresholding mechanism.
+
+**Note:** As mentioned in the [SERVICE.md](https://github.com/WICG/conversion-measurement-api/blob/main/SERVICE.md) doc, we are planning on adding differential privacy noise to the output of the MPC computation. For hierarchical computation, this will require adding noise to each level.
+
+### Example query model
+
+The following is a high level query model that could be achieved with the incremental distributed point function (DPF) primitive.
+
+Imagine the following prefix-tree of counts, where each key is 3 bits long in total (last level of the tree)
 ```
-conversion-signing-url=<URL>
+1-bit prefix counts:       102                   402
+2-bit prefix counts:    99       3           400       2
+3-bit prefix counts: 99   0    1   2      200   200   1  1
 ```
 
-Once an impression / conversion is registered, the browser should interact with the signing endpoint, which should provide authentication tokens that will attach to conversion reports and be verified by the helper servers. Helper servers can (optionally) discard reports that are not signed. This is explored in more detail [here](SERVICE.md#authenticating-inputs).
 
-### Many-per-click / Once-per-click
+We could have a query configuration asking for counts at every prefix level, with some strategy for iterating down and pruning the tree.
+
+
+```json
+{
+  prefix_lengths: [1, 2, 3],
 
-The event-level API allows up to 3 conversions per click, and subsequent ones are dropped. It is reasonable to support at least this number in the aggregate API as well.
+  // assume no noise for simplicity, this setting could allow distributing the
+  // noise across levels in some non-uniform fashion.
+  privacy_budget_per_prefix: [...],
 
-We also propose some new functionality “once-per-click” (OPC), that allows conversions to be dropped if they are duplicate. We’ll modify the conversion redirect to accept an extra param:
+  // Don't expand a node if it doesn't reach the threshold.
+  expansion_threshold_per_prefix: [5, 5, 5]
+}
 ```
-local-dedupe-key=<key>
+
+The output (after merging results from two servers) could look like this, where the aggregate data is returned for all queried prefixes.
+
+
+```json
```

Comment:
We should fix the syntax highlighting here and use strings as the keys.

Note sure what to do about the comment

Reply:
done, removed json syntax highlighting